drivers/bluetooth/hci/h4.c

```diff
@@ -307,6 +337,12 @@ static inline void read_header(void)
 	case H4_ACL:
 		get_acl_hdr();
 		break;
+	case H4_ISO:
+		if (IS_ENABLED(CONFIG_BT_ISO)) {
+			get_iso_hdr();
+			break;
+		}
+		__fallthrough;
```

Comment:
Weird, I thought that should be printing an error no CODE_UNREACHABLE, even the original code is wrong is that regard since for other code path we just print an error and continue.

Reply:
Actually the code is correct, we can't continue if we cannot process the header, so either we continue using CODE_UNREACHABLE or we do process the header and drop it later, but given that ISO never to be enabled by the stack there is really something wrong if we got an ISO packet. @jhedberg since you are the original author do you have any opnion on this? Shall we continue using CODE_UNREACHABLE if we cannot parse the header (disabled CONFIG_BT_ISO) or just parse it and drop it later?

Reply:
Johan won't be able to respond before the feature freeze. It just looks weird but I understand the logic. 
I suppose the CODE_UNREACHABLE is not needed, since there is a return after there. Keeping it or removing it, either is fine with me.

Reply:
I think the idea was to cause __builtin_unreachable which seems to be used by fatal errors, perhaps it would be a better idea to assert in that case.